src/compiler/checker.ts

```diff
@@ -8840,7 +8840,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         node,
                         factory.updateLiteralTypeNode(node.argument, rewriteModuleSpecifier(node, node.argument.literal)),
                         node.attributes,
-                        node.qualifier,
+                        visitNode(node.qualifier, visitExistingNodeTreeSymbols, isEntityName),

```

Comment:
I definitely don't have enough context on the workings of the checker to give any informed feedback, but I can at least ask at a high level do we have some confidence that whether anything more needs to be done to avoid this general class of problem?

Reply:
Tbh, I don't know this part of TS too well either so we need to wait for @weswigham or @dragomirtitian to give some feedback on this. I'd expect that other things in this function (like `node.attributes` here) should be wrapped like this too (if only this is the right fix).

Reply:
I would definitely suspect a lot of this needs this same fix.

Reply:
100% - any child node we copy should have its' children visited - any missing visitor calls on manually remapped nodes are an oversight, presumably.